src/librustc/middle/mem_categorization.rs

```diff
@@ -643,7 +643,13 @@ impl<'a, 'gcx, 'tcx> MemCategorizationContext<'a, 'gcx, 'tcx> {
                 Ok(self.cat_rvalue_node(id, span, expr_ty))
           }
 
-          Def::Static(_, mutbl) => {
+          Def::Static(def_id, mutbl) => {
+            // `#[thread_local]` statics may not outlive the current function.
```

Comment:
The entire file has broken (read: old-skool `match` rules) indentation, I'm actually on a multiple of 4 and the `Def::Static` line above is wrong.